specs/merge/beacon-chain.md

```diff
@@ -209,41 +197,38 @@ def process_block(state: BeaconState, block: BeaconBlock) -> None:
     process_randao(state, block.body)
     process_eth1_data(state, block.body)
     process_operations(state, block.body)
-    # Pre-merge, skip execution payload processing
-    if is_execution_enabled(state, block):
-        process_execution_payload(state, block.body.execution_payload, EXECUTION_ENGINE)  # [New in Merge]
+    process_execution_payload(state, block.body)  # [New in Merge]
 ```
 
-#### Execution payload processing
+### Execution payload processing
 
-##### `process_execution_payload`
+#### `process_execution_payload`
 
 ```python
-def process_execution_payload(state: BeaconState,
-                              execution_payload: ExecutionPayload,
-                              execution_engine: ExecutionEngine) -> None:
-    """
-    Note: This function is designed to be able to be run in parallel with the other `process_block` sub-functions
-    """
-    if is_transition_completed(state):
-        assert execution_payload.parent_hash == state.latest_execution_payload_header.block_hash
-        assert execution_payload.number == state.latest_execution_payload_header.number + 1
-
-    assert execution_payload.timestamp == compute_time_at_slot(state, state.slot)
-
-    assert execution_engine.new_block(execution_payload)
-
+def process_execution_payload(state: BeaconState, body: BeaconBlockBody) -> None:
```

Comment:
did you globally define the execution engine instead of passing it in?

I would push for the other way around, again for testing and clarity of what this function does and requires

Reply:
> did you globally define the execution engine instead of passing it in?

Right, the same as with the `bls` module. This is so for consistency with the other `process_` functions inside `process_block`. I don't feel too strongly.

<img width="363" alt="Screenshot 2021-06-08 at 15 53 46" src="https://user-images.githubusercontent.com/731743/121208095-b779bb00-c871-11eb-8369-9e5bec549819.png">

Reply:
Passing in the engine to where it is used makes testing a lot easier, every global breaks encapsulation and increases the difficulty of reusing the code.

Reply:
Reverted back to passing the execution engine explicitly :) Thanks for the feedback 👍